spec/controllers/admin/users_controller_spec.rb

```diff
@@ -3,164 +3,126 @@
-          expect(json['email']).to eq(current_user.email)
-          expect(json['role']).to eq(current_user.role)
-        end
-      end
+    let(:params) { current_user.slug }
 
-      context 'with invalid params' do
-        let(:params) { 'invalid-slug' }
-
-        it 'returns errors' do
-          subject
+    context 'with valid params' do
+      it 'returns the given user' do
+        subject
 
-          expect(response).to have_http_status :not_found
-          expect(json).to eq({ 'errors' => 'User not found' })
-        end
+        expect(response).to have_http_status :ok
+        expect(json['name']).to eq(current_user.name)
+        expect(json['email']).to eq(current_user.email)
+        expect(json['role']).to eq(current_user.role)
       end
+    end
 
-      context 'when JWT Token is invalid' do
-        let(:jwt_token) { 'invalid-token' }
+    context 'with invalid params' do
+      let(:params) { 'invalid-slug' }
 
-        it 'returns Unauthorized' do
-          subject
+      it 'returns errors' do
+        subject
 
-          expect(response).to have_http_status :unauthorized
-          expect(JSON.parse(response.body)).to include('message' => 'Unauthorized')
-        end
+        expect(response).to have_http_status :not_found
+        expect(json).to eq({ 'errors' => 'User not found' })
       end
     end
 
-    describe '#update' do
-      subject { put(admin_user_path(current_user.slug), headers: { Authorization: jwt_token }, params: params) }
-
-      let(:email) { 'new-foobar@mail.com' }
-      let(:params) do
-        {
-          email: email
-        }
-      end
-
-      context 'with valid params' do
-        it 'updates the given user' do
-          subject
+    include_examples 'Invalid JWT Token'
+  end
 
-          expect(response).to have_http_status :ok
-          expect(json['message']).to eq('User updated with success')
-          expect(current_user.reload.email).to eq('new-foobar@mail.com')
-        end
-      end
+  describe '#update' do
+    subject { put(admin_user_path(current_user.slug), headers: { Authorization: jwt_token }, params: params) }
 
-      context 'with invalid params' do
-        let(:email) { 'second.user@mail.com' }
+    let(:email) { 'new-foobar@mail.com' }
+    let(:params) do
+      {
+        email: email
+      }
+    end
 
-        it 'returns errors' do
-          subject
+    context 'with valid params' do
+      it 'updates the given user' do
+        subject
 
-          expect(response).to have_http_status :unprocessable_entity
-          expect(json['errors']).to include('Email has already been taken')
-        end
+        expect(response).to have_http_status :ok
+        expect(json['message']).to eq('User updated with success')
+        expect(current_user.reload.email).to eq('new-foobar@mail.com')
       end
+    end
 
-      context 'when JWT Token is invalid' do
-        let(:jwt_token) { 'invalid-token' }
+    context 'with invalid params' do
```

Comment:
```suggestion
    context 'with an email that already exists' do
```

aí talvez dá pra adicionar mais um teste com `"with invalid email"` e daí vc seta `let(:email) { 'invalid@email' }`